src/OpenTelemetry/Metrics/Tags.cs

```diff
@@ -20,26 +20,23 @@ namespace OpenTelemetry.Metrics
     {
         private readonly int hashCode;
 
-        public Tags(string[] keys, object[] values)
+        public Tags(KeyValuePair<string, object>[] keyValuePairs)
         {
-            this.Keys = keys;
-            this.Values = values;
+            this.KeyValuePairs = keyValuePairs;
 
-            unchecked
+            var hash = 17;
+            for (int i = 0; i < this.KeyValuePairs.Length; i++)
             {
-                var hash = 17;
-                for (int i = 0; i < this.Keys.Length; i++)
+                unchecked
                 {
-                    hash = (hash * 31) + this.Keys[i].GetHashCode() + this.Values[i]?.GetHashCode() ?? 0;
+                    hash = (hash * 31) + this.KeyValuePairs[i].Key.GetHashCode() + this.KeyValuePairs[i].Value?.GetHashCode() ?? 0;
```

Comment:
Two comments...

1. The hash code computation looks a bit off. I'm not an expert here (would prefer we use [HashCode.Combine](https://learn.microsoft.com/en-us/dotnet/api/system.hashcode.combine?view=net-7.0)) but based on what I have seen each sub component should append with the prime. Fixed up below.

3. Might be able to squeeze a little more perf out of it using a ref to the array item. Demo below.

```csharp
                ref var item = ref this.KeyValuePairs[i];

                hash = (hash * 31) + item.Key.GetHashCode();
                hash = (hash * 31) + item.Value?.GetHashCode() ?? 0;
```

Reply:
Updated this.